real_time/arduino_src/radio_buggy_mega/Makefile

```diff
@@ -328,7 +330,7 @@ ALL_ASFLAGS = -mmcu=$(MCU) -I. -x assembler-with-cpp $(ASFLAGS)
 
 
 # Default target.
-all: begin gccversion sizebefore build sizeafter end
+all: begin gccversion sizebefore fingerprint build sizeafter end
 

```

Comment:
Adding the fingerprint rule here doesn't always tell make to build thing in the right order. Take this case for example:
```
[02:44] [ihartwig@nilla2:radio_buggy_mega]% make program

Compiling: main.cpp
avr-gcc -c -mmcu=atmega2560 -I. -gdwarf-2 -DF_CPU=16000000UL -DUSART0_ENABLED -DUSART2_ENABLED  -Os -funsigned-char -funsigned-bitfields -fpack-struct -fshort-enums -Wall -Wstrict-prototypes -Wa,-adhlns=main.lst  -std=gnu99 -DBUGGY=no_buggy -MD -MP -MF .dep/main.o.d main.cpp -o main.o
cc1plus: warning: command line option '-Wstrict-prototypes' is valid for C/ObjC but not for C++
cc1plus: warning: command line option '-std=gnu99' is valid for C/ObjC but not for C++
In file included from ../lib_avr/uart/uart_extra.h:9:0,
                 from main.cpp:9:
../lib_avr/uart/uart.h:124:0: warning: "USART0_ENABLED" redefined
 #define USART0_ENABLED
 ^
<command-line>:0:0: note: this is the location of the previous definition
main.cpp:13:25: fatal error: fingerprint.h: No such file or directory
 #include "fingerprint.h"
                         ^
compilation terminated.
make: *** [main.o] Error 1
```

Since it is possible that fingerprint.h could be used in any c, cpp, or asm source file, it should be a dependency of all the object file (.o) rules:
```
# Compile: create object files from C source files.
%.o : %.c fingerprint
	@echo
	@echo $(MSG_COMPILING) $<
	$(CC) -c $(ALL_CFLAGS) $< -o $@ 

%.o : %.cpp fingerprint
	@echo
	@echo $(MSG_COMPILING) $<
	$(CC) -c $(ALL_CFLAGS) $< -o $@ 

# Compile: create assembler files from C source files.
%.s : %.c fingerprint
	$(CC) -S $(ALL_CFLAGS) $< -o $@

%.s : %.cpp fingerprint
	$(CC) -S $(ALL_CFLAGS) $< -o $@


# Assemble: create object files from assembler source files.
%.o : %.S fingerprint
	@echo
	@echo $(MSG_ASSEMBLING) $<
	$(CC) -c $(ALL_ASFLAGS) $< -o $@

# Create preprocessed source for use in sending a bug report.
%.i : %.c fingerprint
	$(CC) -E -mmcu=$(MCU) -I. $(CFLAGS) $< -o $@ 

%.i : %.cpp fingerprint
	$(CC) -E -mmcu=$(MCU) -I. $(CFLAGS) $< -o $@ 
```